sample-app/src/components/Navigation.tsx

```diff
@@ -2,18 +2,43 @@ import classNames from 'classnames';
 import { useCallback, useEffect, useLayoutEffect, useRef, useState } from 'react';
 import { NavLink, useLocation } from 'react-router-dom';
 import { ReactComponent as KebabIcon } from '../icons/kebab.svg';
-import '../sass/Navigation.scss';
+import { useComposedCssClasses, CompositionMethod } from '../hooks/useComposedCssClasses';
+
+interface NavigationCssClasses {
+  nav?: string,
+  linksWrapper?: string,
+  menuWrapper?: string,
+  navLink?: string,
+  activeNavLink?: string,
+  menuButton?: string,
+  menuButtonContainer?: string,
+  menuButton___menuOpen?: string,
+  menuContainer?: string
+}
+
+const builtInCssClasses: NavigationCssClasses = {
+  nav: 'border-b border-gray-200 text-gray-600 flex space-x-6 font-medium pt-4',
+  navLink: 'whitespace-nowrap py-4 px-1 font-medium text-md border-b-2 border-opacity-0 hover:border-gray-300',
+  activeNavLink: 'text-blue-600 border-blue-600 border-b-2 border-opacity-100 hover:border-blue-600',
+  menuButtonContainer: 'relative flex flex-grow justify-end mr-4',
+  menuButton: 'flex flex-row items-center font-medium text-md px-1 border-b-2 border-opacity-0 hover:border-gray-300',
+  menuButton___menuOpen: 'bg-gray-200',
+  menuContainer: 'absolute flex flex-col bg-white border top-14 mt-0.5'
+}
 
 interface LinkData {
   to: string,
   label: string
 }
 
 interface NavigationProps {
-  links: LinkData[]
+  links: LinkData[],
+  customCssClasses?: NavigationCssClasses,
+  compositionMethod?: CompositionMethod
```

Comment:
what do you think about `cssCompositionMethod` so it's more specific in terms of what it's use for?

Reply:
sure, I can address that in another PR